src/main/java/com/iexec/common/chain/ChainDeal.java

```diff
@@ -72,15 +73,19 @@ public static DealParams stringToDealParams(String params) {
         try {
             DealParams dealParams = new ObjectMapper().readValue(params, DealParams.class);
             if (dealParams.getIexecInputFiles() == null) {
-                dealParams.setIexecInputFiles(new ArrayList<>());
+                dealParams.setIexecInputFiles(List.of());
+            }
+            if (dealParams.getIexecSecrets() == null) {
+                dealParams.setIexecSecrets(Map.of());
             }
             return dealParams;
         } catch (IOException e) {
             //the requester want to execute one task with the whole string
             return DealParams.builder()
                     .iexecArgs(params)
-                    .iexecInputFiles(new ArrayList<>())
+                    .iexecInputFiles(List.of())
```

Comment:
```suggestion
                    .iexecInputFiles(Collections.emptyList())
```
?
https://stackoverflow.com/a/39400310

Reply:
Fixed in https://github.com/iExecBlockchainComputing/iexec-common/pull/298/commits/c642ed8631c33d38f36a6e45da21850c8272bf56